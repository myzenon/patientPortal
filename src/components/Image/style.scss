@import '~themes/colors.scss';

.imageWrapper  {
    position: relative;
    display: inline-block;
    cursor: pointer;
    border: 0px solid $border-color;
    // height: 205px;
     width: 100%;	
     max-width: 277px;
    border-radius: 6px;
    box-shadow: 0 2px 7px 0 rgba(0,0,0,0.14);
}


.zoomInText{
    position: absolute;
    top: 55%;
    left: 15%;
    height: 30px;
    width: 196px;
    color: #FFFFFF;
    font-family: "Nunito Sans";
    font-size: 22px;
    font-weight: 800;
    line-height: 30px;
    text-align: center;
}


.noImageWrapper  {
    background-color: #344559;
    position: relative;
    display: inline-block;
    border: 0px solid $border-color;
    height: 205px;
    width: 277px;
    border-radius: 6px;
    box-shadow: 0 2px 7px 0 rgba(0,0,0,0.14);
}


.NoImageText{
    position: absolute;
    top: 40%;
    left: 15%;
    color: #fff;
    font-family: "Nunito Sans";
    font-size: 22px;
    font-weight: 800;
    line-height: 30px;
    text-align: center;
}

.imageMagnifyZoomIn {
    position: absolute;
    top: 30%;
    left: 45%;
}

.imageZoomWrapper {
    height: 485px;
    width: 837.33px;
    object-fit: fill;
    position: absolute;
    border: 0px solid $border-color;
    border-radius: 20px;
    padding: 2.5%;
    z-index: 50;
}
