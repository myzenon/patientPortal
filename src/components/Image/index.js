import React, { Component } from 'react'
import style from './style.scss'

import Magnifier from 'assets/enlarge_icon.svg'

import ImagePopup from 'components/ImagePopup'
  
export default class Image extends Component {
    state = {
        zoom: false
    }
    handleZoomIn = () => {
        this.setState({ zoom: true })
    }
    handleZoomOut = () => {
        this.setState({ zoom: false })
    }
    render() {
        if (this.state.zoom) {
            return this.renderZoom()
        }
        return (
            <div >  

             { this.props.src ? ( 
                         <div className={style.imageWrapper}>
                         <img
                            alt=""
                            {...this.props}
                            onClick={this.handleZoomIn}
                        /> 
                        <Magnifier  className={style.imageMagnifyZoomIn}  onClick={this.handleZoomIn}/> 

                       <label className={style.zoomInText}> Make Photo Larger </label>
                        </div>
                    ) : 
                    <div className={style.noImageWrapper}>
                          <label className={style.NoImageText}> Image Not Available</label>
                     </div>
                    
                    }  
            </div>
        )
    }
    renderZoom() {
        const props = {...this.props} 
        return (
            <span>            { this.props.src ? ( 
                <div className={style.imageWrapper}>
                <img
                   alt=""
                   {...this.props}
                   onClick={this.handleZoomIn}
               />  
               </div>
           ) : 
           <div className={style.noImageWrapper}>
                 <label className={style.NoImageText}> Image Not Available</label>
            </div>
           
           }  

            <div className={style.imageZoomWrapper} > 
              {  <ImagePopup  {...props}  onClose={this.handleZoomOut}  /> }
            </div>
            </span>

        )
    }
}