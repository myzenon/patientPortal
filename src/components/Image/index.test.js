import React from 'react'
import Image from './index'

test('renders correctly', () => {
    const wrapper = render (
        <Image />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks on image | image zoom', (done) => {
    const wrapper = mount (
        <Image />
    )
    wrapper.find('img').first().simulate('click')
    setTimeout(() => {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})

test('clicks on magnify | image zoom', (done) => {
    const wrapper = mount (
        <Image />
    )
    wrapper.find('img[alt="Zoom-In"]').simulate('click')
    setTimeout(() => {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})