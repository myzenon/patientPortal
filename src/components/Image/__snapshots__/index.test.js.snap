// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicks on image | image zoom 1`] = `
<div
  class="imageZoomWrapper"
>
  <img
    alt=""
    class="imageZoomImage"
  />
  <img
    alt="Zoom-Out"
    class="imageMagnifyZoomOut"
    src="magnify-out.png"
  />
</div>
`;

exports[`clicks on magnify | image zoom 1`] = `
<div
  class="imageZoomWrapper"
>
  <img
    alt=""
    class="imageZoomImage"
  />
  <img
    alt="Zoom-Out"
    class="imageMagnifyZoomOut"
    src="magnify-out.png"
  />
</div>
`;

exports[`renders correctly 1`] = `
<div
  class="imageWrapper"
>
  <img
    alt=""
  />
  <img
    alt="Zoom-In"
    class="imageMagnifyZoomIn"
    src="magnify-in.png"
  />
</div>
`;
